drivers/flash/flash_stm32_ospi.c

```diff
@@ -0,0 +1,1498 @@
+				} else {
+					/* use the default sector erase cmd */
+					cmd_erase.Instruction =
+						(dev_cfg->data_mode == OSPI_SPI_MODE)
+						? SPI_NOR_CMD_SE /* erase sector size 3 bytes */
+						: SPI_NOR_OCMD_SE;
+					cmd_erase.AddressMode =
+						(dev_cfg->data_mode == OSPI_SPI_MODE)
+						? HAL_OSPI_ADDRESS_1_LINE
+						: HAL_OSPI_ADDRESS_8_LINES;
+					cmd_erase.AddressDtrMode =
+						(dev_cfg->data_rate == OSPI_DTR_TRANSFER)
+						? HAL_OSPI_ADDRESS_DTR_ENABLE
+						: HAL_OSPI_ADDRESS_DTR_DISABLE;
+					cmd_erase.AddressSize = HAL_OSPI_ADDRESS_32_BITS;
+					cmd_erase.Address = addr;
+				}
+			}
+			ospi_send_cmd(dev, &cmd_erase);
+			if (bet != NULL) {
+				addr += BIT(bet->exp);
+				size -= BIT(bet->exp);
+			} else {
+				addr += SPI_NOR_SECTOR_SIZE;
+				size -= SPI_NOR_SECTOR_SIZE;
+			}
+		ret = stm32_ospi_mem_ready(&dev_data->hospi,
+						 dev_cfg->data_mode, dev_cfg->data_rate);
+		}
+	}
+
+	ospi_unlock_thread(dev);
+
+	return ret;
+}
+
+/* Function to read the flash with possible OSPI/SPI and STR/DTR */
+static int flash_stm32_ospi_read(const struct device *dev, off_t addr,
+				 void *data, size_t size)
+{
+	const struct flash_stm32_ospi_config *dev_cfg = DEV_CFG(dev);
+	int ret;
+
+	if (!ospi_address_is_valid(dev, addr, size)) {
+		LOG_DBG("Error: address or size exceeds expected values: "
+			"addr 0x%lx, size %zu", (long)addr, size);
+		return -EINVAL;
+	}
+
+	/* read non-zero size */
+	if (size == 0) {
+		return 0;
+	}
+
+	OSPI_RegularCmdTypeDef cmd = ospi_prepare_cmd(dev_cfg->data_mode, dev_cfg->data_rate);
+	/* Instruction and DummyCycles are set below */
+	cmd.Address = addr;
+	/* DataSize is set by the read cmd */
+
+	LOG_INF("OSPI: read %u data", size);
+	ospi_lock_thread(dev);
+
+	/* configure other parameters */
+	if (dev_cfg->data_rate == OSPI_DTR_TRANSFER) {
+		/* DTR transfer rate (==> Octal mode) */
+		cmd.Instruction = SPI_NOR_OCMD_DTR_RD;
+		cmd.DummyCycles = SPI_NOR_DUMMY_RD_OCTAL_DTR;
+	} else {
+		/* STR transfer rate */
+		if (dev_cfg->data_mode == OSPI_SPI_MODE) {
+			/* SPI and STR : use fast read with addr on 4 bytes */
+			cmd.Instruction = SPI_NOR_CMD_4B_FAST_RD;
+			cmd.DummyCycles = SPI_NOR_DUMMY_RD;
+		} else {
+			/* OPI and STR */
+			cmd.Instruction = SPI_NOR_OCMD_RD;
+			cmd.DummyCycles = SPI_NOR_DUMMY_RD_OCTAL;
+		}
+	}
+
+	ret = ospi_read_access(dev, &cmd, data, size);
+
+	ospi_unlock_thread(dev);
+
+	return ret;
+}
+
+/* Function to write the flash (page program) : with possible OSPI/SPI and STR/DTR */
+static int flash_stm32_ospi_write(const struct device *dev, off_t addr,
+				  const void *data, size_t size)
+{
+	const struct flash_stm32_ospi_config *dev_cfg = DEV_CFG(dev);
+	struct flash_stm32_ospi_data *dev_data = DEV_DATA(dev);
+	int ret = 0;
+
+	if (!ospi_address_is_valid(dev, addr, size)) {
+		LOG_DBG("Error: address or size exceeds expected values: "
+			"addr 0x%lx, size %zu", (long)addr, size);
+		return -EINVAL;
+	}
+
+	/* write non-zero size */
```

Comment:
```suggestion
	/* ignore zero size erase */
```
or even remove the comment as it is quite obvious what the check is for...

Reply:
changing comment